firewood/src/db.rs

```diff
@@ -390,7 +391,8 @@ impl api::Db for Db {
     type Proposal = Proposal;
 
     async fn revision(&self, root_hash: HashKey) -> Result<Arc<Self::Historical>, api::Error> {
-        if let Some(rev) = self.get_revision(&TrieHash(root_hash)) {
+        let rev = block_in_place(|| self.get_revision(&TrieHash(root_hash)));
```

Comment:
💩

Reply:
Why `block_in_place` over `spawn_blocking`. The latter is usually used when one is unsure of the outer context in which the blocking call is made. In other words, this is only faster if there aren't a bunch of other tasks running on the same thread. 

My intuition tells me that performance would be much more predictable using `spawn_blocking` but *could be* faster using `block_in_place`. I'm not necessarily opposed to it, but I think it should be justified.

Reply:
Please respond before next round of review

Reply:
The trouble here is that I can't move `self` into the closure, and `block_in_place` can borrow self since the thread is guaranteed to be done with it before returning.

I think if this is slower or faster is in the eye of the beholder (or of the benchmark tool). Since we don't know, and there are lifetime issues, this seems like a decent choice.